### ISSUE TITLE
Trigger: Handle case where no events are triggered

### ISSUE THREAD

**Issue by tmgreenfield1101** [QuakeMigrate/QuakeMigrate#125]
**Describe the bug**
When running Trigger I receive an UnboundError related to the local variable 'discarded'. The only difference between my script and that of the example is I have removed the optional region argument to the trigger call.

**To Reproduce**
Steps to reproduce the behaviour (including error messages if applicable).
```
(quakemigrate_dev) tg286@circinus:/space/tg286/indonesia/ocean_bottom_data/qmigrate$ python ./trigger.py 
==============================================================================================================
==============================================================================================================
	QuakeMigrate RUN - Path: outputs/runs/test_run - Name: test_run
==============================================================================================================
==============================================================================================================

==============================================================================================================
	TRIGGER - Triggering events from .scanmseed
==============================================================================================================

	Triggering events from 2020-03-25T00:00:00.000000Z to 2020-03-26T00:00:00.000000Z

	Trigger parameters:
		Pre/post pad = 120.0 s
		Marginal window = 1.0 s
		Minimum event interval  = 60.0 s

		Triggering from the normalised maximum coalescence trace.

		Trigger threshold method: static
		Static threshold = 1.45

==============================================================================================================
	Reading in .scanmseed...

	    No .scanmseed file found for day 2020_084!

	    No .scanmseed file found for day 2020_086!
	    Warning! No .scanmseed data found for pre-pad!
	    Warning! No .scanmseed data found for post-pad!
	    ...from 2020-03-25T00:00:00.000000Z - 2020-03-25T23:59:59.900000Z.

	Triggering events...
	No events triggered at this threshold - try a lower detection threshold.

	Plotting trigger summary...
Traceback (most recent call last):
  File "./trigger.py", line 58, in <module>
    trig.trigger(starttime, endtime, interactive_plot=False)
  File "/space/tg286/quake_migrate/tim_area/QuakeMigrate/quakemigrate/signal/trigger.py", line 244, in trigger
    self._trigger_batch(batchstart, batchend, region, interactive_plot)
  File "/space/tg286/quake_migrate/tim_area/QuakeMigrate/quakemigrate/signal/trigger.py", line 299, in _trigger_batch
    data, region, discarded,
UnboundLocalError: local variable 'discarded' referenced before assignment
```

**Expected behavior**

I would have expected trigger to run

**Desktop (please complete the following information):**
 - Operating System: Ubuntu
 - Python version: 3
 - QuakeMigrate version: 1.0

**Additional context**

**Comment by tmgreenfield1101** [comment]
Saw that no events triggered. Maybe best to add a more informative error here?

**Comment by TomWinder** [comment]
I'd class this as a bug still, because it's important to be able to try a high threshold and then inspect the plot to guide your choice of a lower threshold

**Comment by TomWinder** [comment]
Tim can you test the fix I have applied at f03b290, please? I have tested it locally and seems to fix the issue.

**Comment by tmgreenfield1101** [comment]
this fixes the issue on my example - thanks - stay tuned though
